es-app/src/views/gamelist/GridGameListView.cpp

```diff
@@ -48,6 +49,30 @@ void GridGameListView::launch(FileData* game)
 	ViewController::get()->launch(game);
 }
 
+void GridGameListView::remove(FileData *game, bool deleteFile)
+{
+	if (deleteFile)
+		Utils::FileSystem::removeFile(game->getPath());  // actually delete the file on the filesystem
+	FileData* parent = game->getParent();
+	if (getCursor() == game)                     // Select next element in list, or prev if none
+	{
+		std::vector<FileData*> siblings = parent->getChildrenListToDisplay();
+		auto gameIter = std::find(siblings.cbegin(), siblings.cend(), game);
+		int gamePos = (int)std::distance(siblings.cbegin(), gameIter);
+		if (gameIter != siblings.cend())
+		{
+			if ((gamePos + 1) < siblings.size())
+			{
+				setCursor(siblings.at(gamePos + 1));
+			} else if ((gamePos - 1) > 0) {
+				setCursor(siblings.at(gamePos - 1));
+			}
+		}
+	}
+	delete game;                                 // remove before repopulating (removes from parent)
```

Comment:
Can you update the remove method to match [BasicGameListView](https://github.com/RetroPie/EmulationStation/blob/master/es-app/src/views/gamelist/BasicGameListView.cpp#L126)?
It is missing the placeholder code which prevents issues with empty custom collections.

Once that is done and squashed I can merge this.

Reply:
@jrassa I updated the remove function to match BasicGameListView but there is 2 things I don't understand.

First, when do we need this ? When I have an empty custom collection, I just can't "enter" it when I press A on it in the custom collection list. I can't test it if I don't know when it happen.

Second, I noticed there is a problem with the addPlaceholder function in the BasicGameListView, it call the add function like this : 
```c++
mList.add(placeholder->getName(), placeholder, (placeholder->getType() == PLACEHOLDER));
```

And the signature of the method in TextListComponent : 
```c++
void add(const std::string& name, const T& obj, unsigned int colorId);
```

The 3rd parameter mismatch, it give a boolean as third parameter instead of an unsigned int.

Reply:
@pjft can hopefully confirm, but I believe where it is necessary is if you are in a custom collection and remove the last item from it.

The parameter mismatch is not an issue.  The bool to int conversions is implicit.  `false == 0` and `true == 1`.

Reply:
@jrassa Ok its look good to me.

Sorry for the false alert, maybe its my bad english but when I saw "colorId" I was expecting something like 0x42CF75FF not 0 or 1, looking deeper into the code it all make sense.